v2/streaming-data-generator/src/test/java/com/google/cloud/teleport/v2/templates/StreamingDataGeneratorIT.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import static com.google.cloud.teleport.it.artifacts.ArtifactUtils.createGcsClient;
+import static com.google.cloud.teleport.it.artifacts.ArtifactUtils.getFullGcsPath;
+import static com.google.cloud.teleport.it.dataflow.DataflowUtils.createJobName;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.auth.Credentials;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.teleport.it.TestProperties;
+import com.google.cloud.teleport.it.artifacts.Artifact;
+import com.google.cloud.teleport.it.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator.Result;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient.JobInfo;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient.JobState;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient.LaunchConfig;
+import com.google.cloud.teleport.it.dataflow.FlexTemplateClient;
+import com.google.cloud.teleport.v2.templates.StreamingDataGenerator.SinkType;
+import com.google.common.io.Resources;
+import com.google.re2j.Pattern;
+import java.io.IOException;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration test for {@link StreamingDataGenerator}. */
+@RunWith(JUnit4.class)
+public final class StreamingDataGeneratorIT {
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final Credentials CREDENTIALS = TestProperties.googleCredentials();
+  private static final String PROJECT = TestProperties.project();
+  private static final String REGION = TestProperties.region();
+  private static final String SPEC_PATH = TestProperties.specPath();
+
+  private static final String SCHEMA_FILE = "gameevent.json";
+  private static final String LOCAL_SCHEMA_PATH = Resources.getResource(SCHEMA_FILE).getPath();
+
+  private static final String TEST_ROOT_DIR = "streaming-data-generator";
```

Comment:
Use the class name as the dir name?
No need to manually create this variable, dir names guaranteed to be unique, and easier to find the test that created the directory in case you need to clean it up manually or something.

Reply:
Ok. The idea was just to keep the existing directory name, but that isn't really necessary.